### DISCUSSION TITLE
cURL error 28 with a self hotsed fivefilters full-text rss feed

### DISCUSSION THREAD

**Discussion by jacksonmalms** [FreshRSS/FreshRSS#6824] [Q&A]
### Describe the bug

Issue: 

FreshRSS throws the following error when I give it a self hosted address.

![Nbb8m6g](https://github.com/user-attachments/assets/4215c480-f8b1-4602-9e30-9d4df69a6f8a)
![2024-09-21_22-53](https://github.com/user-attachments/assets/cbb0ca0c-79d7-405e-90a1-ee06fa325d56)

Context:

I am self hosting both FreshRSS and FiveFilter's Full-Text RSS docker container in order to get fully displayed pages from articles that do not natively support full-text on their RSS feed. 

When I generate a feed from the Full-Text RSS container, it displays correctly and is accessible from my browser and another RSS reader that I have been using called Feedbro (it is a firefox extension, cringey name I know).

Example from Feedbro:
![2024-09-21_22-43](https://github.com/user-attachments/assets/504d498d-b8d1-47da-ac5f-bfb9feae889a)

Example from browser:
![2024-09-21_22-55](https://github.com/user-attachments/assets/de637b10-ab02-4dc6-8af4-c03e54ddd249)

Now my problem now is that it does not work on Fresh RSS even though both FreshRSS and the Full-Text RSS containers are running on the same host: 
![2024-09-21_22-38](https://github.com/user-attachments/assets/6e549041-1024-470e-b3e8-27cf043e27c0)

When I give it the URL, it is just kaput and throws me this error:
cURL error 28: Connection timed out after 20002 milliseconds [http://192.168.0.103:591/makefulltextfeed.php?url=sec%3A%2F%2Fwww.bleepingcomputer.com%2Ffeed%2F&amp;max=10&amp;links=preserve&amp;exc=&amp;submit=Create+Feed]

So what gives? 

I even gave it the #force_feed flag on my url (in my case: http://192.168.0.103:591/makefulltextfeed.php?url=sec%3A%2F%2Fwww.bleepingcomputer.com%2Ffeed%2F&max=10&links=preserve&exc=&submit=Create+Feed) like the error suggested, but the feed is just not accessible from my FreshRSS instance and I am totally lost at this point.

Help would be greatly appreciated.

Hopefully someday I can be knowledgeable enough to help a noob like myself right now, but right now I am the noob :(

### To Reproduce

1. Generate full-text rss feed from an rss feed
2. Attempt to add it into FreshRSS
3. Error
4. ????

### Expected behavior

It worky

### FreshRSS version

1.24.3 

### Environment information

- Database version: [SQLite]
- PHP version: [PHP 8.1]
- Installation type: [Docker image: lscr.io/linuxserver/freshrss:latest]
-Web server type: [nginx]
- Device: [OrangePi Zero 3]
- OS: [Ubuntu 22.04]
- Browser: [Firefox]

**Comment by jacksonmalms** [comment]
Okay this is not a bug! My issue is caused due to my lack of understanding of how docker/containers work and how communicate to each other (or don't communicate to each other).

I spun up an Ubuntu server VM in my homelab and made a new FreshRSS container to test and see if it was a problem with the containers not being able to communicate on the same host. So my issue is resolved.

But now I have a question for education purposes unrelated to FreshRSS. What if I wanted the two containers on the same host to be able to communicate, how could I do that? I'll probably figure that out on my own but if I don;t maybe someone here knows how and could explain it.

**Comment by Frenzie** [comment]
`docker network create`

there might also be some examples in our docs, possibly using `docker compose` rather than directly.
https://docs.docker.com/engine/network/

**Comment by Alkarex** [comment]
> What if I wanted the two containers on the same host to be able to communicate, how could I do that?

You need to address them by their docker name, like `http://fivefilters-fullfeedrss-1/` and not by IP.
See our [Docker documentation](https://github.com/FreshRSS/FreshRSS/blob/edge/Docker/README.md)
